### PR TITLE
Support insertion/merging of included list/map

### LINKED ISSUE #11: Support inserting elements in an include file to an array/map
Consider this use case:

```yaml
- filter: A
  options: ...
# ... need to insert many filters from an include file here, e.g., a station list file
- filter: Z
  options: ...
```

There is likely to be similar use cases for map/object? E.g.,

```yaml
people:
    Alice:
        greet_phrase: Hello
        # ... insert many key-value elements from an include file here
        is_happy_today: true

### PR DESCRIPTION
**Pull request by matthewrmshin**
Allow insertion of elements read from a list in an include file into a list of the root file.

Ditto for insertion of key-value pairs from a map in an include file into a map of the root file.

Close #11.

### CODE REVIEW

File: docs/data-process.rst [hunk 1]
```diff
@@ -99,6 +99,107 @@ LIMITATIONS
    recognised will not be processed.
 
 
+Modularisation / Include with Merge
+-----------------------------------
+
+A common use case for include is to merge a list read from an include
+file into the current list, (or similarly to merge a map/object read from an
+include file into the current map/object). We can tell the processor with the
+``MERGE`` keyword.
```
Comment (steoxley) [inline docs/data-process.rst:108]: Might be worth continuing this with:
We can tell the processor with the ``MERGE`` keyword, which if not specified defaults to false.

**Review by steoxley — APPROVED** [review]
I've only eyeballed. Looks good. I like the new unit tests!

File: docs/data-process.rst (hunk 1 above)
Comment (matthewrmshin) [reply docs/data-process.rst]: Done.

**Comment by KrissyRayD242** [comment]
Can you mention in the description, that only content will be inserted which is not already in the original yaml? 
For example I had: 

**first.yaml**
```
map:
  human:
    say: Hello World
  others:
    INCLUDE: sayings.yaml
    MERGE: true
  martians:
    say: Greeting Earthlings  
  cat: 
    speak: ['muh','beh']
    young: kitten
```

**sayings.yaml**
```
cat:
  say: ['meow', 'miaow']
dog:
  say: woof woof
```

**result.yaml**
```
map:
  human:
    say: Hello World
  martians:
    say: Greeting Earthlings
  cat:
    speak:
    - muh
    - beh
    young: kitten
  dog:
    say: woof woof
```

**Comment by KrissyRayD242** [comment]
Also, I discovered by accident that when I do:
```
cat:
  speaks: Dutch French
  others:
  INCLUDE: catdata.yaml
  MERGE: True

```
or
```
cat:
  speaks: Dutch French
  INCLUDE: catdata.yaml
  MERGE: True

```
I'm getting:
```
chase:
- rodents
- birds
like:
- food
- play
```

**Comment by KrissyRayD242** [comment]
Other than that looks good, I went through the code and tested it.

**Review by KrissyRayD242 — APPROVED** [review]

**Comment by KrissyRayD242** [comment]
If I put the `others` section after `cat`, then cat gets overwritten:
```
map:
  human:
    say: Hello World
  martians:
    say: Greeting Earthlings  
  cat: 
    speak: ['muh','beh']
    young: kitten
  others:
    INCLUDE: sayings.yaml
    MERGE: true
```
result:
```
map:
  human:
    say: Hello World
  martians:
    say: Greeting Earthlings
  cat:
    say: meow miaow
  dog:
    say: woof woof woof woof
```

Might be worth adding some more comments to explain cases like that. :)

**Comment by matthewrmshin** [comment]
@KrissyRayD242 Thanks.

I am going to have a look at the first issue, as I have expected the values in the include file to override. I may have got the logic wrong somewhere.

The second issue is the correct behaviour, as the logic is expected to replace the orginal key-value in place. Not sure what can be done to improve the logic. More checking? Otherwise, we'll probably have to reply on a schema to report the issue.

**Comment by KrissyRayD242** [comment]
works now :)

**Pull request merged**